per_hostname_tls_settings.go

```diff
@@ -0,0 +1,117 @@
+package cloudflare
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+)
+
+// HostnameTLSSetting represents the metadata for a user-created tls setting.
+type HostnameTLSSetting struct {
+	Hostname  string      `json:"hostname"`
+	Value     interface{} `json:"value"`
```

Comment:
this is a bit of a smell for me and will cause you issues in other tooling like terraform. do we need to accept multiple types here? can we settle on one?

Reply:
API input can be float, string, or list of strings (and the response will be string or list of strings for tls setting value)
The float part doesn't matter because the int input can be a string too.
But the list of strings cannot be sent in as a string to the API. It must be an actual list.
http2 setting can be "on" or "off"
min_tls_version setting can be float or string representations of 1.0, 1.1, 1.2, 1.3
ciphers setting must be a list of strings such as ["AES128-GCM-SHA256","ECDHE-RSA-AES128-GCM-SHA256"]

It's the ciphers input that makes this tricky. I see no other way other than to make this be an interface - at least if we want it to follow how the API works. For the Go library, I guess we can just always make it be a string. And I can make the func then convert that string into a list of ciphers. Would this be better?

Customer would then always input a string and the library converts it to what it needs to be (in the case of ciphers), and converts the API response ciphers list into a string.